lisa/sut_orchestrator/libvirt/transformers.py

```diff
@@ -0,0 +1,523 @@
+        self._build_and_install(code_path)
+        return _get_package_version(self._node, "libvirtd")
+
+
+class CloudHypervisorSourceInstaller(CloudHypervisorInstaller):
+
+    __distro_package_mapping = {
+        Ubuntu.__name__: ["gcc"],
+        CBLMariner.__name__: ["gcc", "binutils", "glibc-devel"],
+    }
+
+    @classmethod
+    def type_name(cls) -> str:
+        return "source"
+
+    @classmethod
+    def type_schema(cls) -> Type[schema.TypedSchema]:
+        return SourceInstallerSchema
+
+    def _build_and_install(self, code_path: PurePath) -> None:
+        result = self._node.execute(
+            "cargo build --release",
+            shell=True,
+            sudo=False,
+            cwd=code_path,
+        )
+        assert not result.exit_code, "Failed to build cloud-hypervisor"
+        self._node.execute(
+            "cp ./target/release/cloud-hypervisor /usr/local/bin",
+            shell=True,
+            sudo=True,
+            cwd=code_path,
+        )
+        self._node.execute(
+            "chmod a+rx /usr/local/bin/cloud-hypervisor",
+            shell=True,
+            sudo=True,
+        )
+        self._node.execute(
+            "setcap cap_net_admin+ep /usr/local/bin/cloud-hypervisor",
+            shell=True,
+            sudo=True,
+        )
+
+    def _install_dependencies(self) -> None:
+        linux: Linux = cast(Linux, self._node.os)
+        packages_list = self.__distro_package_mapping[type(linux).__name__]
+        linux.install_packages(list(packages_list))
+        result = self._node.execute(
+            "curl https://sh.rustup.rs -sSf | sh -s -- -y",
+            shell=True,
+            sudo=False,
+        )
+        assert not result.exit_code, "Failed to install Rust & Cargo"
+        self._node.execute("source ~/.cargo/env", shell=True, sudo=False)
+        if isinstance(self._node.os, Ubuntu):
+            output = self._node.execute("echo $HOME", shell=True)
+            path = self._node.get_pure_path(output.stdout)
+            self._node.execute(
+                "cp .cargo/bin/* /usr/local/bin/",
+                shell=True,
+                sudo=True,
+                cwd=path,
+            )
+        self._node.execute("cargo --version", shell=True)
+
+    def install(self) -> str:
+        runbook: SourceInstallerSchema = self.runbook
+        self._log.info("Installing dependencies for Cloudhypervisor...")
+        self._install_dependencies()
+        self._log.info("Cloning source code of Cloudhypervisor ...")
+        code_path = _get_source_code(runbook, self._node, self.type_name(), self._log)
+        self._log.info("Building source code of Cloudhypervisor...")
+        self._build_and_install(code_path)
+        return _get_package_version(self._node, "cloud-hypervisor")
+
+
+class CloudHypervisorReleaseInstaller(CloudHypervisorInstaller):
+    @classmethod
+    def type_name(cls) -> str:
+        return "latest_release"
```

Comment:
It doesn't need to specify "latest", maybe it needs to support different versions later. How about name it "binary"?

Reply:
Sure, will rename it to "release_binary", but current code supports to get only the latest release binary.

Reply:
The `binary` should be enough.

Reply:
changed.